### PR TITLE
Staging sites: remove `onError` callbacks from query hooks

### PR DESCRIPTION
**Pull request by flootr**
Preparatory PR for the upcoming `@tanstack/query` v5 upgrade.

## Proposed Changes

PR removes usages of `onError` callbacks on `useQuery` hooks. They're removed with v5 of the library.

## Testing Instructions

To properly test this you need one AT site without a staging site and one AT site with a staging site. In any case you should visit `/hosting-config` and verify the **Staging site card** is behaving as described below. You also need to use the _Network request blocking_ feature of your browser devtools (or use your sandbox to throw errors but I think the former is easier).

The `staleTime` for those queries is set to 10 seconds so either block the URL before you navigate to `/hosting-config` or wait 10 seconds and focus in and out of the window to trigger `refetchOnWindowFocus`. Alternatively, use React Query Devtools to invalidate the query and trigger a refetch.

### AT site without staging site

- Block network requests to `/validate-quota`
- After 3 unsuccessful request attempts you should see an error message which looks like this:

Screenshot 2023-11-09 at 16 43 39

### AT site with staging site

- Block network requests to `/staging-site`
- After 3 unsuccessful request attempts you should see an error message which looks like this:

Screenshot 2023-11-09 at 16 47 44

#### Staging site

This can be a bit tricky to test because the error is only shown when there's no cached data. So make sure to remove the data from the query cache (e.g. via React Query Devtools) before you follow the steps:

- Block network requests to `/production-site-details`
- After 3 unsuccessful request attempts you should see an error message which looks like this:

Screenshot 2023-11-09 at 16 51 39

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/hosting/staging-site-card/use-has-valid-quota.ts [hunk 1]
```diff
@@ -3,7 +3,9 @@ import wp from 'calypso/lib/wp';
 
 export const USE_VALID_QUOTA_QUERY_KEY = 'valid-quota';
 
-export const useHasValidQuotaQuery = ( siteId: number, options: UseQueryOptions ) => {
+type HasValidQuotaOptions = Pick< UseQueryOptions, 'enabled' >;
```
Comment (flootr) [inline client/my-sites/hosting/staging-site-card/use-has-valid-quota.ts:6]: This is a drive-by change. We were never exposing the complete `UseQueryOptions` so we shouldn't type it like that. Same goes for the other two query hooks which we change in this PR.

**Review by wojtekn — APPROVED** [review]
Thanks for the cleanup @flootr . The code change looks good, and all cases work as expected.

> You also need to use the Network request blocking feature of your browser devtools (or use your sandbox to throw errors but I think the former is easier).

Great testing instructions! TIL about the Network request blocking feature.
